File: LeetCode/21-merge-sorted-list.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def __init__(self):
        self.head = None
        self.tail = None
    def mergeTwoLists(self, lista, listb):
        arra = []
        print(lista)
        tempa = lista
        tempb = listb
        while(tempa):
            arra.append(tempa.val)
            tempa = tempa.next
        while (tempb):
            arra.append(tempb.val)
            tempb = tempb.next
        arra.sort()
        for el in arra:
            self.push(el)
        return self.head
      
    def push(self, value):
        newNode = ListNode(value)
        if self.head is None:
            self.head = newNode
            self.tail = newNode
        else:
            self.tail.next = newNode
            self.tail = newNode
        

File: LeetCode/21-merge-sorted-list.ts
class Solution {
    head: ListNode | null;
    tail: ListNode | null;

    constructor() {
        this.head = null;
        this.tail = null;
    }

    mergeTwoLists(lista: ListNode | null, listb: ListNode | null): ListNode | null {
        let arra: number[] = [];
        
        let tempa: ListNode | null = lista;
        let tempb: ListNode | null = listb;

        while (tempa) {
            arra.push(tempa.val);
            tempa = tempa.next;
        }

        while (tempb) {
            arra.push(tempb.val);
            tempb = tempb.next;
        }

        arra.sort((a, b) => a - b); // Sort numerically

        for (let el of arra) {
            this.push(el);
        }

        return this.head;
    }

    push(value: number): void {
        let newNode = new ListNode(value);
        if (this.head === null) {
            this.head = newNode;
            this.tail = newNode;
        } else {
            if (this.tail) { // Type guard to ensure tail is not null
                this.tail.next = newNode;
            }
            this.tail = newNode;
        }
    }
}